sql/table_stats.cc

```diff
@@ -441,7 +441,7 @@ void fill_table_stats_cb(const char *db,
   copy_comp_stats_with_races(&stats->comp_stats, comp_stats);
 
   stats->n_lock_wait.inc(n_lock_wait);
-  stats->n_lock_wait_timeout.inc(n_lock_wait_timeout);
+  stats->n_lock_wait_timeout.set_maybe(n_lock_wait_timeout);
```

Comment:
One issue I can see for this is that it may break InnoDB stats, since they also go through this code path.